frontend/src/components/CourseSearchList.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect } from "react";
+import { useAppDispatch, useAppSelector } from "../app/hooks";
+import CourseCard from "./CourseCard";
+import { Pagination } from "react-headless-pagination";
+import {
+  fetchCourseInfosByPage,
+  fetchFCEInfos,
+  selectCourseResults,
+} from "../app/courses";
+import { ChevronLeftIcon, ChevronRightIcon } from "@heroicons/react/outline";
+import Loading from "./Loading";
+
+const CoursePage = () => {
+  const dispatch = useAppDispatch();
+
+  const pageCourses = useAppSelector((state) => state.courses.pageCourses);
+  const page = useAppSelector((state) => state.courses.page);
+
+  const exactResultsCourses = useAppSelector(
+    (state) => state.courses.exactResultsCourses
+  );
+  const exactMatchesOnly = useAppSelector(
+    (state) => state.user.filter.exactMatchesOnly
+  );
+
+  const showFCEs = useAppSelector((state) => state.user.showFCEs);
+  const showCourseInfos = useAppSelector((state) => state.user.showCourseInfos);
+  const loggedIn = useAppSelector((state) => state.user.loggedIn);
+
+  let coursesToShow: string[] = [];
+
+  if (exactMatchesOnly) {
+    coursesToShow = [...exactResultsCourses];
+  } else if (page === 1 && exactResultsCourses.length > 0) {
+    coursesToShow = [
+      ...exactResultsCourses,
+      ...pageCourses.filter(
+        (courseID) => !exactResultsCourses.includes(courseID)
+      ),
+    ];
+  } else {
+    coursesToShow = [...pageCourses];
+  }
+
+  const results = useAppSelector(selectCourseResults(coursesToShow));
+
+  useEffect(() => {
+    console.log(coursesToShow);
+    if (loggedIn && coursesToShow) {
+      dispatch(fetchFCEInfos({ courseIDs: coursesToShow }));
+    }
+  }, [coursesToShow.join(" "), loggedIn]);
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'coursesToShow' and 'dispatch'. Either include them or remove the dependency array.](https://app.codacy.com/gh/ScottyLabs/course-tool/pullRequest?prid=9667658)
```suggestion
  }, [coursesToShow, dispatch, loggedIn]);
```